compiler/src/iree/compiler/Codegen/Common/EncodingUtils.cpp

```diff
@@ -20,76 +20,9 @@ using IREE::Encoding::EncodingAttr;
 using IREE::Encoding::getEncodingAttr;
 using IREE::Encoding::getEncodingContractionDims;
 
-// If tensorType has the encoding of a matmul RESULT with narrow N, returns
-// the transposed type. Otherwise, just returns tensorType.
-static RankedTensorType transposeIfNarrowNResult(RankedTensorType tensorType) {
```

Comment:
I was expecting to see this code copied to some CPU-specific file, but I can't see it? Or is it being not needed anymore as we have a different way of performing the equivalent logic?

Reply:
Oh I see, this is what you are explaining in the PR description. Wow --- this is fantastic! I'm just now realizing the power of the refactoring you've been doing. Being able to simply transpose the encoding because it hasn't been materialized yet in a RankedTensorType. Amazing!

Reply:
Side note: this is making transpose-narrow-N so neat that it removes much of the objection to doing it on GPU, if a need arises -- such as maybe with GPU ukernels.

Reply:
It is not needed because it is handled by `getEncodingInfo` method. Firstly, we get the encoding info. Secondly, we transpose the encoding info inplace if it is a narrow_n case. I think the reason of constructing the ranked tensor type with the new encoding is that we want to infer the same encoding info. This logic can be handled by the attribute, and we do not need to expose the logic to the type converter.

Reply:
yep, what you said. you beat me to this. :D